docs/documentation/comparison_of_click_and_rich_click.md

```diff
@@ -0,0 +1,109 @@
+
+The above are the same as importing from `click`.
+
+**rich-click**'s subclasses all have the word Rich in front of them!
+
+- `from rich_click import RichCommand`
+- `from rich_click import RichGroup`
+- `from rich_click import RichContext`
+
+### Echo and interactive elements
+
+**rich-click** deliberately does _not_ enrich certain Click features:
+
+- `click.echo()`
+- `click.echo_via_pager()`
+- `click.confirm()`
+- `click.prompt()`
+
+You are free to use these functions and they are available via `import rich_click as click`,
+but Rich's markup will not work with these functions because these functions are just the base Click implementations,
+without any changes.
+
+This is a deliberate decision that we are unlikely to change in the future.
+We do not want to maintain a more spread-out API surface, and we encourage users to become comfortable using Rich directly; it's a great library and it's worth learning a little bit about it!
+If you'd like Rich markup for your echos and interactive elements, then you can:
+
+- Replace `click.echo()` with `rich.print()`
+- Replace `click.echo_via_pager()` with `rich.Console().pager()`
+- Replace `click.confirm()` with `rich.prompt.Confirm.ask()`
+- Replace `click.prompt()` with `rich.prompt.Prompt.ask()`
+
+Below is a side-by-side comparison of Click and Rich implementations of echos and interactive elements in **rich-click**:
+
+=== "Click"
+
+    ```python
+    import rich_click as click
+
+    @click.command("greet")
+    def greet():
+        name = click.prompt(click.style("What is your name?", fg="blue"))
+    
+        if not click.confirm(click.style("Are you sure?", fg="blue")):
+            click.echo(click.style("Aborting", fg="red"))
+            return
+    
+        click.echo(click.style(f"Hello, {name}!", fg="green"))
+    
+    if __name__ == "__main__":
+        greet()
+    ```
+
+=== "Rich"
+
+    ```python
+    import rich_click as click
+    import rich
+    from rich.prompt import Confirm, Prompt
+
+    @click.command("greet")
+    def greet():
+        name = Prompt.ask("[blue]What is your name?[/]")
+    
+        if not Confirm.ask("[blue]Are you sure?[/]"):
+            rich.print("[red]Aborting[/]"))
+            return
+    
+        rich.print(f"[green]Hello, {name}![/]")
+    
+    if __name__ == "__main__":
+        greet()
+    ```
+
+## Additional rich-click features
+
+- **rich-click** has a configuration object, `RichHelpConfiguration()`, that allows for control over how **rich-click** help text renders, so you are not just locked into the defaults. More information about this is described in [the **Configuration** docs](configuration.md).
+- **rich-click** comes with a CLI tool that allows you to convert regular Click CLIs into **rich-click** CLIs, and also lets you render your **rich-click** CLI help text as HTML and SVG. More information about this is described in [the **rich-click CLI** docs](rich_click_cli.md).
+
+**rich-click** is a slim library.
```

Comment:
Feels a bit stranded here? Delete?

Reply:
Oops! I forgot what I was typing there, but it's not necessary.